ecl/hql/hqlattr.cpp

```diff
@@ -4112,31 +4116,46 @@ bool canDefinitelyProcessWithTranslator(IHqlExpression * record)
                                 return false;
                         }
                     }
+                    if (!checkCanDefinitelyProcessWithTranslator(cur->queryRecord()))
+                        return false;
                     break;
                 }
             }
             break;
         }
         case no_ifblock:
             {
+                if (!checkCanDefinitelyProcessWithTranslator(cur->queryChild(1)))
+                    return false;
                 IHqlExpression * cond = cur->queryChild(0);
                 node_operator condOp = cond->getOperator();
                 //Match the subset of the expressions that are supported by the record translation
                 //false positives are acceptable, false negatives are not
-                if ((condOp == no_eq) || (condOp == no_ne))
+                if ((condOp == no_eq) || (condOp == no_ne) || (condOp == no_in) || (condOp == no_notin))
```

Comment:
Add a couple more examples to the positives (because of examples which were now caught by the other fixes)